api/oc_network_events.c

```diff
@@ -57,14 +57,21 @@ oc_process_network_event(void)
   OC_LIST_LOCAL(network_events);
   oc_list_copy(network_events, g_network_events);
   oc_list_init(g_network_events);
-#ifdef OC_NETWORK_MONITOR
+ #ifdef OC_NETWORK_MONITOR
   bool interface_up = g_interface_up;
-  g_interface_up = false;
+   g_interface_up = false;
   bool interface_down = g_interface_down;
   g_interface_down = false;
 #endif /* OC_NETWORK_MONITOR */
   oc_network_event_handler_mutex_unlock();
 
+ #ifdef OC_DYNAMIC_ALLOCATION 
+  if(oc_list_length(network_events) >= OC_MAX_NUM_CONCURRENT_REQUESTS) {
```

Comment:
need to call `oc_get_network_events_queue_length(endpoint->device)` without locking (`oc_get_network_events_queue_length_locked(endpoint->device)` ???) instead of `oc_list_length`

Reply:
Done. Implemented non-blocking method to determine the number of events for a specific device
`static size_t get_events_queue_length(size_t device, oc_list_t events)`